File: api/contrib/signXML/module.php
<?php
/** @file module.php
 * A brief file description.
 * A more elaborated file description.
 */

/** \addtogroup Core 
 *  @{
 */

/**
 * \defgroup Module
 * @{
 */


/**
 * Boot up procedure
 */
function signXML_bootMeUp(){
	// Just booting up
}

/**
 * Init function
 */

function signXML_init(){    
	$paths = array(
		array(
			'r' => 'signFE',
			'action' => 'signFE',
			'access' => 'users_openAccess', 
			'access_params' => 'accessName',			
			'params' => array(
				array("key" => "p12Url", "def" => "", "req" => true),
				array("key" => "pinP12", "def" => "", "req" => true),
				array("key" => "inXml", "def" => "", "req" => false),							
				array("key" => "tipodoc", "def" => "", "req" => true)
			),			
			'file' => 'sign.php'
		)
	);
        
	return $paths;
}


/**************************************************/
//In the access you can use users_openAccess if you want anyone can use the function
// or users_loggedIn if the user must be logged in
/**************************************************/



/**
 * Get the perms for this module
 */
function MODULENAME_access(){

	$perms = array(
		array(
			# A human readable name
			'name'        => 'Do something with this module',
			# Something to remember what it is for
			'description' => 'What can be achieved with this permission',
			# Internal machine name, no spaces, no funny symbols, same rules as a variable
			# Use yourmodule_ prefix
			'code'        => 'mymodule_access_one',
			# Default value in case it is not set
			'def'        => false, //Or true, you decide
		),
	);

}

/**@}*/
/** @}*/

File: migration/python-api/auto_port/api/contrib/ejemplo/module.py
"""
AUTO-PORTED FROM: api/contrib/ejemplo/module.php
Mode: mechanical baseline (no manual refactor).
"""

from __future__ import annotations

def ejemplo_bootMeUp():
    """
    AUTO-PORTED PHP BODY (verbatim)
    --------------------------------
    
        // Just booting up
    """
    raise NotImplementedError('Auto-port placeholder: complete behavior port here.')

def ejemplo_init():
    """
    AUTO-PORTED PHP BODY (verbatim)
    --------------------------------
    
        $paths = array(
            array(
                'r'             => 'hola',
                'action'        => 'module_hola',
                'access'        => 'users_openAccess',
                'access_params' => 'accessName',
                'file'          => 'ejemplo.php'
            ),
            array(
                'r'             => 'un_usuario',
                'action'        => 'unUsuario',
                'access'        => 'users_openAccess',
                'access_params' => 'accessName',
                'params'        => array(
                    array("key" => "nombre",    "def" => "", "req" => true),
                    array("key" => "apellido",  "def" => "", "req" => true)
                ),
                'file'          => 'ejemplo.php'
            )
        );
    
        return $paths;
    """
    raise NotImplementedError('Auto-port placeholder: complete behavior port here.')

def MODULENAME_access():
    """
    AUTO-PORTED PHP BODY (verbatim)
    --------------------------------
    
        $perms = array(
            array(
                # A human readable name
                'name'        => 'Do something with this module',
                # Something to remember what it is for
                'description' => 'What can be achieved with this permission',
                # Internal machine name, no spaces, no funny symbols, same rules as a variable
                # Use yourmodule_ prefix
                'code'        => 'mymodule_access_one',
                # Default value in case it is not set
                'def'         => false, // Or true, you decide
            ),
        );
    """
    raise NotImplementedError('Auto-port placeholder: complete behavior port here.')
